src/MSBuild/XMake.cs

```diff
@@ -3218,14 +3221,31 @@ private static ILogger[] ProcessLoggingSwitches
             // Add any loggers which have been specified on the commandline
             distributedLoggerRecords = ProcessDistributedLoggerSwitch(distributedLoggerSwitchParameters, verbosity);
 
-            ProcessConsoleLoggerSwitch(noConsoleLogger, consoleLoggerParameters, distributedLoggerRecords, verbosity, cpuCount, loggers);
+            // Choose default console logger
+            bool outputSupportsFancyLogger = !Console.IsOutputRedirected && // Avoid using the FancyLogger when output is redirected to a file
+                Environment.GetEnvironmentVariable("TERM") != "dumb"; // Avoid using FancyLogger when output is dumb (does not support ANSI). TODO: Check for better ways of figuring out terminals' capabilities
```

Comment:
IMO you should have separate log messages for each of these, with basically the content of the comments you have here. That way, if a user expected the fancy logger, they can see exactly why we chose not to use it.